src/random/random_internal.h

```diff
@@ -0,0 +1,36 @@
+/* -*- mode: C -*-  */
+/*
+   IGraph library.
+   Copyright (C) 2021 The igraph development team
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc.,  51 Franklin Street, Fifth Floor, Boston, MA
+   02110-1301 USA
+
+*/
+
+#ifndef IGRAPH_RANDOM_INTERNAL_H
+#define IGRAPH_RANDOM_INTERNAL_H
+
+#include "igraph_types.h"
+
+__BEGIN_DECLS
+
+IGRAPH_PRIVATE_EXPORT igraph_error_t igraph_random_sample_real(
```

Comment:
`IGRAPH_PRIVATE_EXPORT` is used when we need to have some internal functions exported from shared libraries so that they can be used in test programs. Functions which are truly purely internal to igraph (i.e. not even used from the tests) should not be marked for exporting at all.